CSharpMath/Structures/Dictionary.cs

```diff
@@ -30,92 +30,60 @@ public class ProxyAdder<TKey, TValue> : IEnumerable {
     public void Add(TKey key1, TKey key2, TKey key3, TValue value) {
       Add(key1, value); Add(key2, value); Add(key3, value);
     }
-    public void Add(TKey key1, TKey key2, TKey key3, TKey key4, TValue value) {
-      Add(key1, value); Add(key2, value); Add(key3, value); Add(key4, value);
-    }
-    public void Add(TKey key1, TKey key2, TKey key3, TKey key4, TKey key5, TValue value) {
-      Add(key1, value); Add(key2, value); Add(key3, value); Add(key4, value);
-      Add(key5, value);
-    }
-    public void Add(TKey key1, TKey key2, TKey key3, TKey key4, TKey key5, TKey key6, TValue value) {
-      Add(key1, value); Add(key2, value); Add(key3, value); Add(key4, value);
-      Add(key5, value); Add(key6, value);
-    }
-    public void Add(TKey key1, TKey key2, TKey key3, TKey key4, TKey key5, TKey key6, TKey key7, TValue value) {
-      Add(key1, value); Add(key2, value); Add(key3, value); Add(key4, value);
-      Add(key5, value); Add(key6, value); Add(key7, value);
-    }
-    public void Add(TKey key1, TKey key2, TKey key3, TKey key4, TKey key5, TKey key6, TKey key7, TKey key8, TValue value) {
-      Add(key1, value); Add(key2, value); Add(key3, value); Add(key4, value);
-      Add(key5, value); Add(key6, value); Add(key7, value); Add(key8, value);
-    }
```

Comment:
These should be re-added for consistency. We shouldn't arbitrarily limit the use of collection initializers.

Reply:
I went back up to 4 although it's not good practice. The 4-method is unused.